src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -55,6 +55,7 @@ public partial class DynamoModel : INotifyPropertyChanged, IDisposable, IEngineC
         private readonly PathManager pathManager;
         private WorkspaceModel currentWorkspace;
         private Timer backupFilesTimer;
+        private Dictionary<Guid, string> backupFilesDict = new Dictionary<Guid, string>();
```

Comment:
The logic here is trying to avoid backing up the same files which have already been backed up. For example for the following case:
1). There are two workspaces and they are backed up;
2). A new dyf workspace is created;
3). When the backing up operation happens again, the first two workspaces should not be backed up again because they have already backed up;
4). For the first two workspaces, we do not know the full paths where they have been backed up. So we need to figure out the pathes from the workspaces' IDs. 

This is why the dictionary is required.